libraries/AP_Filesystem/AP_Filesystem_Sys.cpp

```diff
@@ -166,8 +170,12 @@ int32_t AP_Filesystem_Sys::read(int fd, void *buf, uint32_t count)
         return -1;
     }
     struct rfile &r = file[fd];
-    count = MIN(count, r.str->get_length() - r.file_ofs);
-    memcpy(buf, &r.str->get_string()[r.file_ofs], count);
+    if (r.is_crash_dump) {
```

Comment:
I suggest we do this:
 - add set_data_const(const uint8_t *data, uint32_t size) to ExpandingString
 - that would mark as "don't free" in ExpandingString destructor
 - add a get_storage_ptr() fn to hal.storage for getting storage ptr
it means no more special cases

Reply:
ok, will do.